efs/main.tf

```diff
@@ -32,3 +32,117 @@ resource "aws_efs_file_system_policy" "file_system_policy" {
   file_system_id = aws_efs_file_system.file_system.id
   policy         = templatefile("${path.module}/templates/${var.policy}.json.tpl", { file_system_arn = aws_efs_file_system.file_system.arn })
 }
+
+resource "aws_security_group" "mount_target_sg" {
+  name        = "mount-target-security-group"
+  description = "Mount target security group"
+  vpc_id      = data.aws_vpc.current.id
+
+  ingress {
+    description     = "EFS"
+    from_port       = 2049
+    to_port         = 2049
+    protocol        = "tcp"
+    security_groups = [aws_security_group.ecs_run_efs.id, aws_security_group.allow_efs_lambda.id]
+  }
+
+  egress {
+    protocol    = "-1"
+    from_port   = 0
+    to_port     = 0
+    cidr_blocks = ["0.0.0.0/0"]
+  }
+
+  tags = merge(
+    var.common_tags,
+    map("Name", "mount-target-outbound-only")
+  )
+}
+
+resource "aws_security_group" "ecs_run_efs" {
```

Comment:
Should this be conditional? Do we always want to add ECS access when setting up an EFS store?

Same question for the Lambda security group below.

Reply:
It should be, I couldn't think of a clean way of doing it with conditionals in the efs module without a load of nested ternary statements so I've changed it round a bit. So now:

* The download lambda has its own security group defined in `lambda/download_files.tf`
* The file format lambda has its own security group defined in `lambda/file_format.tf`
* The file format build ecs task has its own security group defined in `ecs/file_format_build.tf`
* Each of the above 3 security groups are output from their modules.
* These security groups are passed as input to the efs module.

This way, we can pass arbitrary security groups to the EFS module to determine what has access to it. This way, if we want other resources other than lambda and ECS accessing EFS, we won't have to change the EFS module.

Also, it's better for each lambda to have its own security group I think so we can have more control over what is allowed to connect to the mount point.

Reply:
That's a neat solution, thanks!